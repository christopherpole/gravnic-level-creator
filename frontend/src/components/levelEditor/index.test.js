import React from 'react';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { LevelEditor } from './index';

configure({ adapter: new Adapter() });

describe('The level editor', () => {
  let props;

  beforeEach(() => {
    props = {
      previewing: false,
    };
  });

  it('Renders without exploding', () => {
    const levelEditor = shallow(<LevelEditor {...props} />);

    expect(levelEditor).toHaveLength(1);
  });

  it('Matches the current snapshot', () => {
    const levelEditor = shallow(<LevelEditor {...props} />);

    expect(toJson(levelEditor)).toMatchSnapshot();
  });
});
