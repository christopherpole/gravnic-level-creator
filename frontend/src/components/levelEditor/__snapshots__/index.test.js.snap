// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LevelEditor /> Matches the current snapshot 1`] = `
<styled.section>
  <styled.div>
    <styled.div>
      <Connect(Component) />
      <Connect(Component) />
    </styled.div>
    <styled.div>
      <Connect(Component) />
      <Connect(Component) />
    </styled.div>
  </styled.div>
</styled.section>
`;
